onchain/cairo/src/tokens/erc20_mintable.cairo

```diff
@@ -83,8 +83,13 @@ mod ERC20Mintable {
     #[abi(embed_v0)]
     impl IERC20MintableImpl of super::IERC20Mintable<ContractState> {
         fn mint(ref self: ContractState, recipient: ContractAddress, amount: u256) {
-            self.ownable.assert_only_owner();
+            // self.ownable.assert_only_owner();
```

Comment:
@MSghais, owner should be set to vault contract address right?

Reply:
Yes let's add an Access Control init in the constructor to the ERC20 Mintable, and also a set_role to add others role grant access, check example here:

// Admin
// Add OPERATOR role to the Vault escrow
// #[external(v0)]
// fn set_control_role(
//     ref self: ContractState, recipient: ContractAddress, role: felt252, is_enable: bool
// ) {
//     self.accesscontrol.assert_only_role(ADMIN_ROLE);
//     assert!(
//         role == ADMIN_ROLE
//             || role == OPERATOR_ROLE // Think and Add others roles needed on the protocol
//             ,
//         "role not enable"
//     );
//     if is_enable {
//         self.accesscontrol._grant_role(role, recipient);
//     } else {
//         self.accesscontrol._revoke_role(role, recipient);
//     }
// }